apps/mark-scan/fai-100-controller/src/fai_100_controllerd.rs

```diff
@@ -381,14 +387,22 @@ fn handle_status_response(
     // Update the currently pressed button so subsequent identical status reports won't trigger a second keypress
     current_status.button_pressed = new_button;
 
+    // Track if enough time has elapsed since positive connection status.
+    // This mitigates an issue where plugging in headphones to the PAT
+    // jack will send status that a PAT device is connected and a sip
+    // and/or puff event has happened.
+    let connection_signal_delay_elapsed = current_status
+        .sip_puff_device_connection_time
+        .is_some_and(|t| t.elapsed() > Duration::from_secs(2));
+
     // Only check for sip & puff actions when the device is connected because
-    // sip/puff values are inverted when no device is connected, per comment above.
-    // Even if values were consistent, logically no sip/puff signal can be sent without a
-    // connected device.
+    // sip/puff values are inverted when no device is connected.
+    // Even if values were consistent, logically no sip/puff signal can be sent without a connected device.
     if current_status.sip_puff_device_connected == SipAndPuffDeviceStatus::Connected {
```

Comment:
I originally made a single `connection_signal_delay_elapsed` check here but the effect was the stray keypresses were only delayed, not ignored. We could increase the delay, but gating just the `send_keystroke` calls while keeping state updates allows us to use a shorter delay (for the benefit of actual snp users) while still solving the problem.